src/crystal/system/print_error.cr

```diff
@@ -3,16 +3,18 @@ module Crystal::System
   # This is useful for error messages from components that are required for
   # IO to work (fibers, scheduler, event_loop).
   def self.print_error(message, *args)
-    print_error(message, *args) do |bytes|
-      {% if flag?(:unix) || flag?(:wasm32) %}
-        LibC.write 2, bytes, bytes.size
-      {% elsif flag?(:win32) %}
-        LibC.WriteFile(LibC.GetStdHandle(LibC::STD_ERROR_HANDLE), bytes, bytes.size, out _, nil)
-      {% end %}
-    end
+    printf(message, *args) { |bytes| print_error(bytes) }
+  end
+
+  def self.print_error(bytes : Bytes) : Nil
+    {% if flag?(:unix) || flag?(:wasm32) %}
+      LibC.write 2, bytes, bytes.size
+    {% elsif flag?(:win32) %}
+      LibC.WriteFile(LibC.GetStdHandle(LibC::STD_ERROR_HANDLE), bytes, bytes.size, out _, nil)
+    {% end %}
```

Comment:
**thought:** I suppose we could consider to parameterize the target file descriptor in a fllow-up. We already have constants like `Crystal::System::FileDescriptor::STDERR`.

Reply:
Good point. Let's use `Crystal::System::FileDescriptor::STDERR_HANDLE` right now :+1:

Reply:
I'd love to have both print to stdout and stderr but then we need a new name? or a couple methods?

Reply:
Oh no, I can use `STDERR_HANDLE` on UNIX because it's a number literal, but it's not a literal on Windows and we may call print_error _before_ we had a chance to initialize the constant (happens all the time on my tracing branch).